### PR TITLE
ios ISO15693 extended read/write commands (0x30 and 0x31)

### PR DESCRIPTION
**Pull request by engeen**
ios ISO15693 extended read/write commands (0x30 and 0x31)
ios 13.0 only available

### CODE REVIEW

File: ios/Classes/SwiftFlutterNfcKitPlugin.swift [hunk 1]
```diff
@@ -379,6 +399,70 @@ public class SwiftFlutterNfcKitPlugin: NSObject, FlutterPlugin, NFCTagReaderSess
         tag = nil
     }
 
+
+
+@available(iOS 13.0, *)
```
Comment (jiegec) [inline ios/Classes/SwiftFlutterNfcKitPlugin.swift:404]: Please reformat the code

File: ios/Classes/SwiftFlutterNfcKitPlugin.swift [hunk 2]
```diff
@@ -379,6 +399,70 @@ public class SwiftFlutterNfcKitPlugin: NSObject, FlutterPlugin, NFCTagReaderSess
...
+
+func getFlutterError(_ arg: Error) -> FlutterError {
+  return FlutterError(code: "\((arg as NSError).code)", message:arg.localizedDescription, details: nil)
+}
+
+//Call to method 'getFlutterError' in closure requires explicit use of 'self' to make capture semantics explicit
+///Users/yuriy/dev/flutter_nfc_kit/ios/Classes/SwiftFlutterNfcKitPlugin.swift:428:23
+
+
+  @available(iOS 13.0, *)
```
Comment (jiegec) [inline ios/Classes/SwiftFlutterNfcKitPlugin.swift:424]: What will happen if it is not available?

File: ios/Classes/SwiftFlutterNfcKitPlugin.swift [hunk 3]
```diff
@@ -379,6 +399,70 @@ public class SwiftFlutterNfcKitPlugin: NSObject, FlutterPlugin, NFCTagReaderSess
...
+
+        if case let .iso15693(tag) = tag {
+            tag.extendedReadSingleBlock(requestFlags: requestFlags, blockNumber: blockNumber) { dataBlock, error in
+                if let error = error {
+                result(self.getFlutterError(error))
+                } else {
+                result(dataBlock)
+                }
+            }
+        }
```
Comment (jiegec) [inline ios/Classes/SwiftFlutterNfcKitPlugin.swift:437]: What if tag is not iso15693?

File: lib/flutter_nfc_kit.dart [hunk 4]
```diff
@@ -344,7 +376,7 @@ class FlutterNfcKit {
   }
 
   /// iOS only, change currently displayed NFC reader session alert message with [message].
-  /// 
+  ///
```
Comment (jiegec) [inline lib/flutter_nfc_kit.dart:379]: If you want to format code, please move these changes to a separate pull request.

**Review by jiegec — CHANGES_REQUESTED** [review]

**Comment by Harry-Chen** [comment]
@engeen Would you mind refining your code according to the reviewer's comments? Then I will be happy to merge your code.

**Comment by engeen** [comment]
Hi, Shegqi. Yeah, There were really good comments and I hope I will find a
time to make a good contribution. But theres a lot of work, so still on
hold.

Вс, 29 окт. 2023 г. в 22:23, Shengqi Chen ***@***.***>:

**Review by Harry-Chen — APPROVED** [review]
LGTM

**Pull request merged**